hudi-client/hudi-java-client/src/main/java/org/apache/hudi/table/action/commit/BaseJavaCommitActionExecutor.java

```diff
@@ -268,6 +269,10 @@ public Iterator<List<WriteStatus>> handleInsert(String idPfx, Iterator<HoodieRec
       LOG.info("Empty partition");
       return Collections.singletonList((List<WriteStatus>) Collections.EMPTY_LIST).iterator();
     }
+    if (table.getIndex().canIndexLogFiles()) {
+      return new JavaLazyInsertIterable<>(recordItr, true, config, instantTime, table, idPfx,
+          taskContextSupplier, new AppendHandleFactory());
+    }
```

Comment:
Should this be put into `BaseJavaDeltaCommitActionExecutor`?  COW with commit action should not use append handle.